includes/course-theme/class-sensei-course-theme-quiz.php

```diff
@@ -0,0 +1,170 @@
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initializes the class.
+	 */
+	public function init() {
+		if ( 'quiz' !== get_post_type() ) {
+			return;
+		}
+
+		$this->maybe_add_quiz_results_notice();
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @access private
+	 *
+	 * @return string The block HTML.
+	 */
+	public function maybe_add_quiz_results_notice() {
+
+		$lesson_id   = Sensei_Utils::get_current_lesson();
+		$quiz_id     = \Sensei()->lesson->lesson_quizzes( $lesson_id );
+		$user_id     = get_current_user_id();
+		$quiz_status = Sensei_Utils::user_lesson_status( $lesson_id, $user_id )->comment_approved;
+
+		// If not one of the statuses that we handle then bail.
+		if ( ! in_array( $quiz_status, [ 'ungraded', 'graded', 'passed', 'failed' ], true ) ) {
+			return '';
+		}
+
+		// Prepare title.
+		$grade = Sensei_Quiz::get_user_quiz_grade( $lesson_id, $user_id );
+		$title = sprintf(
+			// translators: The placeholder is the quiz grade.
+			__( 'Your Grade: %1$s%%', 'sensei-lms' ),
+			$grade
+		);
+		if ( 'ungraded' === $quiz_status ) {
+			$title = __( 'Awaiting grade', 'sensei-lms' );
+		}
+
+		// Prepare message.
+		$text = __( "You've passed the quiz and can continue to the next lesson.", 'sensei-lms' );
+		if ( 'ungraded' === $quiz_status ) {
+			$text = __( 'Your answers have been submitted and your teacher will grade this quiz shortly.', 'sensei-lms' );
+		} elseif ( 'failed' === $quiz_status ) {
+			$passmark = absint( get_post_meta( $quiz_id, '_quiz_passmark', true ), 2 );
+			$text     = sprintf(
+				// translators: The first placeholder is the minimum grade required, and the second placeholder is the actual grade.
+				__( 'You require %1$s%% to pass this quiz. Your grade is %2$s%%.', 'sensei-lms' ),
+				$passmark,
+				$grade
+			);
+		}
+
+		$actions = [];
+
+		// Prepare complete lesson button.
+		$should_show_complete_lesson = true;
+		if ( 'failed' === $quiz_status || Sensei_Utils::user_completed_lesson( $lesson_id, $user_id ) ) {
+			$should_show_complete_lesson = false;
+		} elseif ( 'ungraded' === $quiz_status && get_post_meta( $quiz_id, '_pass_required', true ) ) {
+			$should_show_complete_lesson = false;
+		}
+		if ( $should_show_complete_lesson ) {
+			$actions[] = self::render_complete_lesson();
+		}
+
+		// Prepare reset quiz button.
+		$reset_allowed = Sensei_Quiz::is_reset_allowed( $lesson_id );
+		if ( $reset_allowed ) {
+			$actions[] = self::render_reset_quiz();
+		}
+
+		// Prepare contact teacher button.
+		$actions[] = self::render_contact_teacher();
+
+		$notices = \Sensei_Context_Notices::instance( 'course_theme_quiz_grade' );
+		$notices->add_notice( 'course-theme-quiz-grade', $text, $title, $actions );
+	}
+
+	/**
+	 * Renders the complete lesson button.
+	 */
+	public static function render_complete_lesson() {
```

Comment:
Do these 3 methods need to be `public`? Or could we convert it to `private` methods?

Reply:
You're right. Converted them to private. 1dbac216ea54b8c0c1e61dca01c7b86edebffd8d